server/routes/routerRequestHandlers/retrieveAllGoalsForPrisoner.ts

```diff
@@ -0,0 +1,21 @@
+import { NextFunction, Request, RequestHandler, Response } from 'express'
+import { EducationAndWorkPlanService } from '../../services'
+import asyncMiddleware from '../../middleware/asyncMiddleware'
+
+/**
+ *  Middleware function that returns a Request handler function to look up the prisoner's goals and store in res.locals
+ */
+const getAllGoalsForPrisoner = (educationAndWorkPlanService: EducationAndWorkPlanService): RequestHandler => {
```

Comment:
Yep, this is exactly right; but you will need to do a unit test for it.
Should be pretty simple; base it on the unit test of `retrieveCuriousInPrisonCourses` 😁